src/Publisher.js

```diff
@@ -28,7 +28,7 @@ module.exports = class Publisher {
             throw new NotReadyError('Server not ready. Please try again shortly.')
         }
 
-        const streamrBinaryMessage = new StreamrBinaryMessage(
+        const streamrBinaryMessage = StreamrBinaryMessageFactory.fromValues(
```

Comment:
I think internally we should always write messages in the newest version. The `Publisher` can well be opinionated about what version messages to publish. Now we would write unsigned messages as version 28 and signed ones as version 29, as per the logic in `StreamrBinaryMessageFactory.fromValues(...)`.

Instead of using the factory I would just construct version 29 messages here: `new StreamrBinaryMessageV29(...)`. In fact I would get rid of `StreamrBinaryMessageFactory.fromValues(...)` completely if this is the only place where it is used! 

For parsing existing (potentially old) byte serializations, we need to be able to choose the right version dynamically. For constructing new serializations however, we are free to decide what version we are constructing!